src/coreclr/pal/src/include/pal/context.h

```diff
@@ -31,7 +31,16 @@ extern "C"
 /* A type to wrap the native context type, which is ucontext_t on some
  * platforms and another type elsewhere. */
 #if HAVE_UCONTEXT_T
-#include <sys/ucontext.h>
+#if defined(__APPLE__) && !defined(_XOPEN_SOURCE)
```

Comment:
It seems it would be better to define `_XOPEN_SOURCE` unconditionally in eng/native/configureplatform.cmake. 
It also needs to be defined in src/pal/src/configure.cmake (`set(CMAKE_REQUIRED_DEFINITIONS -D_XOPEN_SOURCE)`) when detecting HAVE_UCONTEXT_T.